### ISSUE TITLE
map(os.path.basename, ...): Argument 1 to "map" has incompatible type overloaded function; expected "Callable[[str], AnyStr]"

### ISSUE THREAD

**Issue by jdufresne** [python/mypy#9864]
```
$ python --version
Python 3.9.1
$ mypy --version
mypy 0.790
```

Minimal code example:

```py
import os

filenames = map(os.path.basename, ["hello/world.txt"])
```

Actual result:

```
$ mypy ~/test.py
/home/jon/test.py:3: error: Argument 1 to "map" has incompatible type overloaded function; expected "Callable[[str], AnyStr]"
Found 1 error in 1 file (checked 1 source file)
```

Expected: No error. This seems like an acceptable use of `os.path.basename`.

---

I'm happy to lend a hand to resolve this issue, but it is not clear to me how to fix it. Is this an issue with `os.path.basename` or `map` or something else?

**Comment by hauntsaninja** [comment]
I think this is the same as https://github.com/python/mypy/issues/5738#issuecomment-427326745 (one of my least favourite mypy issues)

**Comment by hauntsaninja** [comment]
Also possibly related: https://github.com/python/mypy/issues/5738#issuecomment-427326745

**Comment by AlexWaygood** [comment]
Similar to #2389, but I'm not sure that it's a dupe.

**Comment by ilevkivskyi** [comment]
This now typechecks correctly.
